src/gui/src/components/query-bar/query-token.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react'
+import { cva } from 'class-variance-authority'
+import { cn } from '@/lib/utils.js'
+import type { VariantProps } from 'class-variance-authority'
+import type { ParsedSelectorToken } from '@vltpkg/query'
+
+type TokenType = ParsedSelectorToken['type']
+
+export const tokenVariantClasses = {
+  pseudo: {
+    light: 'text-purple-700 after:bg-purple-600/10',
+    dark: 'dark:text-purple-400 after:dark:bg-purple-600/20',
+  },
+  attribute: {
+    light: 'text-blue-700 after:bg-blue-600/10',
+    dark: 'dark:text-blue-400 after:dark:bg-blue-600/15',
+  },
+  class: {
+    light: 'text-green-700 after:bg-green-600/10',
+    dark: 'dark:text-green-400 after:dark:bg-green-600/15',
+  },
+  id: {
+    light: 'text-pink-700 after:bg-pink-600/10',
+    dark: 'dark:text-pink-400 after:dark:bg-pink-600/15',
+  },
+  combinator: {
+    light: 'text-gray-700 after:bg-gray-600/10',
+    dark: 'dark:text-gray-400 after:dark:bg-gray-600/20',
+  },
```

Comment:
Let's support these types here too (feel free to tweak their colors):

```suggestion
  },
  comment: {
    light: 'text-gray-500 after:bg-gray-400/10',
    dark: 'dark:text-gray-400 after:dark:bg-gray-600/20',
  },
  universal: {
    light: 'text-gray-700 after:bg-gray-600/10',
    dark: 'dark:text-gray-400 after:dark:bg-gray-600/20',
  },
  nesting: {
    light: 'text-gray-700 after:bg-gray-600/10',
    dark: 'dark:text-gray-400 after:dark:bg-gray-600/20',
  },
  string: {
    light: 'text-gray-700 after:bg-gray-600/10',
    dark: 'dark:text-gray-400 after:dark:bg-gray-600/20',
  },
```